flink-libraries/flink-cep-scala/src/main/scala/org/apache/flink/cep/scala/conditions/Context.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.cep.scala.conditions
+
+import java.io.Serializable
+
+/**
+  * The context used when evaluating the {@link IterativeCondition condition}.
+  */
+trait Context[T] extends Serializable {
+  /**
+    * @return An { @link Iterable} over the already accepted elements
+    *                    for a given pattern. Elements are iterated in the order they were
+    *                    inserted in the pattern.
+    * @param name The name of the pattern.
+    */
+  def apply(name: String): Iterable[T]
+}
```

Comment:
Why having different method names between Java and Scala? In Java this method is called `getEventsForPattern()`. We could rename both to sth like `getEventsForState()` if you agree, but I would suggest that same methods should have the same name and also, specifically for this, writing `context(stateName)` does not really seem nice. If you agree I can change it and then merge.